core/sample.h

```diff
@@ -21,11 +21,11 @@ struct sample                         // BASE TYPE BYTES  UNITS    RANGE
 	depth_t depth;                    // int32_t    4    mm     (0-2000 km)            dive depth of this sample
 	depth_t stopdepth;                // int32_t    4    mm     (0-2000 km)            depth of next deco stop
 	temperature_t temperature;        // uint32_t   4    mK     (0-4 MK)               ambient temperature
-	pressure_t pressure[MAX_SENSORS]; // int32_t    4    mbar   (0-2 Mbar)             cylinder pressures (main and CCR o2)
+	pressure_t pressure[MAX_SENSORS]; // int32_t  2x4    mbar   (0-2 Mbar)             cylinder pressures (main and CCR o2)
 	o2pressure_t setpoint;            // uint16_t   2    mbar   (0-65 bar)             O2 partial pressure (will be setpoint)
-	o2pressure_t o2sensor[3];         // uint16_t   6    mbar   (0-65 bar)             Up to 3 PO2 sensor values (rebreather)
+	o2pressure_t o2sensor[3];         // uint16_t 3x6    mbar   (0-65 bar)             Up to 3 PO2 sensor values (rebreather)
 	bearing_t bearing;                // int16_t    2  degrees  (-1 no val, 0-360 deg) compass bearing
-	uint8_t sensor[MAX_SENSORS];      // uint8_t    1  sensorID (0-254)                ID of cylinder pressure sensor
+	int16_t sensor[MAX_SENSORS];      // int16_t  2x2  sensorID (0-16k)                ID of cylinder pressure sensor
 	uint16_t cns;                     // uint16_t   1     %     (0-64k %)              cns% accumulated
 	uint8_t heartbeat;                // uint8_t    1  beats/m  (0-255)                heart rate measurement
 	volume_t sac;                     //            4  ml/min                          predefined SAC
```

Comment:
would be nice to update the total size of the structure below. And man, it's huge. Given that we have a LOT of them in a large dive log with multiple dive computers, potentially short sample rate
Given that no one dives with more than 30 tanks (even if you look at the extreme dives to 300m), why aren't 127 sensors enough? Isn't this overkill?

Reply:
> would be nice to update the total size of the structure below. 

I didn't see that. Could you check my calculations - I don't think that they were correct before.

> Given that no one dives with more than 30 tanks (even if you look at the extreme dives to 300m), why aren't 127 sensors enough? Isn't this overkill?

They should of course be enough. Especially given that the highest tank that is entered here is the first O2 cylinder. I was just thinking, let's cover even pathological cases. This discussion is why I was reluctant with this PR. Feel free to replace it.

Reply:
I think the better answer would be to go to `int8_t`.
If someone needs more than 127 sensors they are so outside of everything that we have ever tested... How would that even work on the profile?
No, let's just do `int_8` here

Reply:
Nitpick: it would be two sensors, but >127 cylinders.

Reply:
well, but don't we support an arbitrary number of cylinders? So should this be to 128bit integers for the people who have more than 4 billion cylinders with them?

Reply:
clearly, this was too snarky of a response. my apologies

more seriously - where would we want to take this? 
I *think* that `int8_t` would be the right answer. And that there might be value in restricting ourselves to 63 cylinders because at that point so much of our infrastructure already breaks. Visualization is shite. The deco algorithms really aren't validated at all for the types of dive where you'd use more than half a dozen cylinders, and certainly once you are in the middle double digits all of this is way outside what we know how to do...
and by going to a limit of 63 for actual dives, the funky "twice the number of cylinders" requirement of the planner is still easily managed.

Thoughts?

Or should I just stop bike shedding and over thinking and merge this?